### DISCUSSION TITLE
A mapping to show hints both for URLs and hyperlinks

### DISCUSSION THREAD

**Discussion by 097115** [kovidgoyal/kitty#6171] [Q&A]
Hi!

Is it possible to create a mapping that would launch hints **both** for URLs (mapped to `ctrl+shift+e` by default) and hyperlinks, like those produced by `ls --hyperlink=always`, etc (mapped to `ctrl+shift+p>y` by defaullt)?

**Comment by kovidgoyal** [comment]
Not builtin, but you can write your own custom matching logic as
described in:
https://sw.kovidgoyal.net/kitty/kittens/hints/#completely-customizing-the-matching-and-actions-of-the-kitten

**Comment by 097115** [comment]
Thanks!

Was actually looking into it before but still can't figure out how to make it to capture both URLs and hyperlinks? Probably, should somehow tinker `re.finditer(r'\w+', text)`? But how? :)

**Comment by kovidgoyal** [comment]
It probably wont be possible, this is because the custom processor is
passed text with the escape codes removed so there is no way for it to
match against hyperlinks.

Your best bet is to submit a PR against the kitten to add an option to
have the kitten add hyperlinks to the found matches, regardless of match
type. Note that this is somewhat subtle since regular matches can overlap
with hyperlink matches.
